segmentstore/server/src/main/java/io/pravega/segmentstore/server/WriterSegmentProcessor.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines a general Segment Processor for the Storage Writer.
+ */
+public interface WriterSegmentProcessor extends AutoCloseable {
+
+    @Override
+    void close();
+
+    /**
+     * Gets a value indicating whether the SegmentAggregator is closed (for any kind of operations).
+     */
+    boolean isClosed();
+
+    /**
+     * Gets the SequenceNumber of the first operation that is not fully committed to Storage.
```

Comment:
Could we add some more detail to the javadoc of this method and others in this interface? For example, define sequence number and explain what property we are trying to preserve by exposing this information to the `StorageWriter`.

Reply:
Sequence Number is universally known in the Segment Container and all operations have it. This is not overriding its definition hence it does not need extra clarification.

Reply:
I really do not like to copy paste comments for the same reason why I don't like to copy-paste code. Once you change it in one place you must remember to change it everywhere else, otherwise any number of bad things can happen. If I define SNs in the container's Javadoc, then define them again here, if I find myself needing to alter the description in the container's Javadoc then I must remember that I also wrote it here too. If I don't, then someone will likely get confused and end up doing something they shouldn't do.

This is not an external API on top of a black/grey box. It's an internal contract so anyone who wishes to write this will likely have access to the full code of the Segment Store core, and will need to deeply understand how it works. Sequence Numbers are one of those building blocks that are needed to comprehend the behavior of anything this low in the API stack.

Reply:
Note that I also said that you could add a pointer, I understand the problems involved in duplicating definitions. Up to you, I think it would improve clarity.